### ISSUE TITLE
Zone transfers of the reverse zone generates forward zone content

### ISSUE THREAD

**Issue by ajf** [coredns/coredns#4063]
**What happened**:

When doing a zone transfer from CoreDNS for a Kubernetes `in-addr.arpa` zone, the forward zone contents are returned:

```
dnsnode1 :: ~ » dig -t AXFR @localhost -p 5300 0.168.192.in-addr.arpa. | awk '/IN/ {print $4}' | sort | uniq -c
  137 A
      4 AAAA
      1 CNAME
      2 SOA
  150 SRV
```

**What you expected to happen**:

That the resulting zone file should have PTR records for mapping 192.168.0.0/24 addresses back to service names, and no other records (besides SOA).

**How to reproduce it (as minimally and precisely as possible)**:

Create a kubernetes plugin (kubernetai in this case) and try to zone transfer it:

```
    kubernetai kube.services.local 192.168.0.0/24 {
      endpoint https://kube-api.local
      pods verified
      endpoint_pod_names
      transfer to 127.0.0.1 ::1
      tls ...
      fallthrough in-addr.arpa ip6.arpa
    }
```

**Environment**:

- the version of CoreDNS: 1.6.0
- OS (e.g: `cat /etc/os-release`):  CentOS 7.6

**Comment by chrisohaver** [comment]
This would be an issue in _kubernetes_ as well - although this bug report is for _kubernetai_, which has zone transfer caveats (the answer only includes records from the first kubernetai instance - the answer is not an aggregation of all kubernetai instance records).

We should probably at least make k8s reject reverse zone transfer requests for now, and then think about how to support them.

**Comment by miekg** [comment]
fix via #4142
